tests/components/modbus/test_modbus_cover.py

```diff
@@ -143,9 +142,13 @@ async def test_register_cover(hass, regs, expected):
     assert state == expected
 
 
-async def test_unsupported_config(hass):
+async def test_unsupported_config(hass, caplog):
     """When user tries to run this component without discovery_info, print an error."""
-    with mock.patch.object(cover, "_LOGGER") as mock_logger:
-        await async_setup_platform(hass, {}, None, None)
-        await hass.async_block_till_done()
-        assert mock_logger.error.called
+    caplog.set_level(logging.WARNING)
+    caplog.clear()
+
+    await async_setup_platform(hass, {}, None, None)
```

Comment:
Set up the cover integration with `async_setup_component` instead.

Reply:
OK

Reply:
@janiversen I want to rework my test to use `async_setup_component`, so I decided to use your harness, where I inititialize `Cover` with an unsupported, legacy config. I have the test ready, but I need to extend the harness.

You always expect that the `base_test` will pass, but in my case, this coroutine needs to fail. In the code, you call `pytest.fail("CONFIG failed, see output")`, and I can't bypass this inside Pytest. It considers the test as failed.

I was thinking to add an optional attribute to the harness, `expectFail = True/False` so that we can control this behavior. Because it's your code and it's the base of all test, I wanted to ask if I can change the code.

I could always copy your code into my test, but in this case it makes sense to me to keep the functionality in the core suite. Later, when we remove the legacy support, this code can test whether our components are not initialized. Basically, what I do in this PR for `Cover`.